extractor/build.gradle

```diff
@@ -33,4 +33,20 @@ test {
         showStandardStreams = true
         events "passed", "skipped", "failed"
     }
-}
\ No newline at end of file
+}
+
+// Custom task to change script permissions and run the setup script
+task setupCheckstyleRepo(type: Exec) {
+    doFirst {
+        // Change permissions to make the script executable
+        file('../setup.sh').setExecutable(true)
```

Comment:
Make file executable before creation of commit and will not need to do this at all

Reply:
> Make file executable before creation of commit and will not need to do this at all

i didn't understand this. But, i am not having this for CI. this is meant for local. So, that whenver some clones checkstyle and runs the build, the checkstyle would be cloned automatically without asking permission. but, i guess if i undertood you right, you are saying make it executable and then commit. If the PR gets merged and someone clones and runs build, won't it ask for permisison to execute the setup.sh?

Reply:
Ok, ignore it. I will do it after merge

Reply:
I have done it though. I guess we wont require pemission.